### PR TITLE
SDT-94: Implement Additional Consumers module unit tests

### PR DESCRIPTION
**Pull request by ladamson-hmcts**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/SDT-94

### Change description ###
Increased unit test coverage for consumers module

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: consumers/src/unit-test/java/uk/gov/moj/sdt/consumers/IndividualRequestConsumerTest.java [hunk 1]
```diff
@@ -211,6 +247,57 @@ void processIndividualRequestOutage ()
...
+     * Disables rethrowOnFailureToConnect to trigger the second if/else statement in super.handleClientErrors,
+     * this triggers the Thread.sleep loop to keep reattempting handleClientErrors
+     * Using Future to cancel out of the Thread.sleep loop for coverage
+     */
+    @Test
+    void processIndividualRequestOutageReattemptLoop() throws SOAPException, InterruptedException {
+        when(mockTransformer.transformDomainToJaxb(individualRequest)).thenReturn(individualRequestType);
+
+        final WebServiceException wsException = new WebServiceException ();
+        wsException.initCause (new ConnectException("Server down error"));
```
Comment (kiran-yenigala-hmcts) [inline consumers/src/unit-test/java/uk/gov/moj/sdt/consumers/IndividualRequestConsumerTest.java:261]: you can use mock object for WebServiceException as well

File: consumers/src/unit-test/java/uk/gov/moj/sdt/consumers/IndividualRequestConsumerTest.java [hunk 2]
```diff
@@ -211,6 +247,57 @@ void processIndividualRequestOutage ()
...
+        doReturn(mockClient).when(individualRequestConsumer)
+            .getClient(anyString(), anyString(), anyString(), anyLong(), anyLong());
+
+        // Disable exception throw to continue past super.handleClientErrors and trigger Thread.sleep
+        individualRequestConsumer.setRethrowOnFailureToConnect(false);
+
+        // Runs the method for half the CONNECTION_TIME_OUT time to trigger InterruptedException on Thread.sleep
+        ExecutorService executor = Executors.newSingleThreadExecutor();
+        Future future = executor.submit(
+            () -> individualRequestConsumer.processIndividualRequest(individualRequest, CONNECTION_TIME_OUT,
```
Comment (kiran-yenigala-hmcts) [inline consumers/src/unit-test/java/uk/gov/moj/sdt/consumers/IndividualRequestConsumerTest.java:274]: do we really need to do the threading stuff in unit testing?

File: consumers/src/unit-test/java/uk/gov/moj/sdt/consumers/IndividualRequestConsumerTest.java [hunk 3]
```diff
@@ -211,6 +247,57 @@ void processIndividualRequestOutage ()
...
+                                                                     RECEIVE_TIME_OUT);
+            fail("Expecting an OutageException here.");
+        } catch (final OutageException toe) {
+            assertTrue(true, GOT_THE_EXCEPTION_AS_EXPECTED);
+        }
+
+
+        verify(mockTransformer, times(2)).transformDomainToJaxb(individualRequest);
+        verify(mockClient, times(3)).submitIndividual(individualRequestType);
+        assertTrue(true, TEST_FINISHED_SUCCESSFULLY);
```
Comment (kiran-yenigala-hmcts) [inline consumers/src/unit-test/java/uk/gov/moj/sdt/consumers/IndividualRequestConsumerTest.java:298]: we don't need this assertTrue

File: consumers/src/unit-test/java/uk/gov/moj/sdt/consumers/IndividualRequestConsumerTest.java [hunk 4]
```diff
@@ -311,4 +398,29 @@ private IndividualResponseType generateResponse() {
...
+        final uk.gov.moj.sdt.ws._2013.sdt.targetapp.indvresponseschema.HeaderType headerType =
+            new uk.gov.moj.sdt.ws._2013.sdt.targetapp.indvresponseschema.HeaderType();
+        headerType.setSdtRequestId("Test");
+
+        responseType.setHeader(headerType);
+
+        final CreateStatusType status = new CreateStatusType();
+        status.setCode(CreateStatusCodeType.ACCEPTED);
+
+        ErrorType mockError = new ErrorType();
```
Comment (kiran-yenigala-hmcts) [inline consumers/src/unit-test/java/uk/gov/moj/sdt/consumers/IndividualRequestConsumerTest.java:416]: mockError -> errorType

**Review by kiran-yenigala-hmcts — COMMENTED** [review]
Comments added

**Review by hemantt — APPROVED** [review]

**Review by kiran-yenigala-hmcts — APPROVED** [review]

**Pull request merged**